apps/design/backend/src/worker/worker.ts

```diff
@@ -1,45 +1,46 @@
 import { extractErrorMessage, sleep } from '@votingworks/basics';
 
-import { BackgroundTask, Store } from '../store';
+import { Workspace } from '../workspace';
+import { processBackgroundTask } from './tasks';
 
-async function processBackgroundTask(
-  _store: Store, // eslint-disable-line @typescript-eslint/no-unused-vars
-  _task: BackgroundTask // eslint-disable-line @typescript-eslint/no-unused-vars
-): Promise<void> {
-  // Simulate a long-running operation
-  // TODO(arsalan): Implement actual processing logic
-  await sleep(3000);
-}
+export async function processNextBackgroundTaskIfAny({
+  log = console.log, // eslint-disable-line no-console
+  workspace,
+}: {
+  log?: (text: string) => void;
+  workspace: Workspace;
+}): Promise<void> {
+  const { store } = workspace;
 
-async function processBackgroundTasks(store: Store): Promise<void> {
-  // eslint-disable-next-line no-constant-condition
-  while (true) {
-    const nextTask = store.getOldestQueuedBackgroundTask();
+  const nextTask = store.getOldestQueuedBackgroundTask();
 
-    if (!nextTask) {
-      await sleep(1000);
-      continue;
-    }
+  if (!nextTask) {
+    await sleep(1000);
+    return;
+  }
 
-    store.startBackgroundTask(nextTask.id);
-    process.stdout.write(`⏳ Processing background task ${nextTask.id}...\n`);
-    try {
-      await processBackgroundTask(store, nextTask);
-    } catch (error) {
-      const errorMessage = extractErrorMessage(error);
-      store.completeBackgroundTask(nextTask.id, errorMessage);
-      process.stdout.write(
-        `❌ Error processing background task ${nextTask.id}:\n${errorMessage}\n`
-      );
-      continue;
-    }
-    store.completeBackgroundTask(nextTask.id);
-    process.stdout.write(
-      `✅ Finished processing background task ${nextTask.id}\n`
+  store.startBackgroundTask(nextTask.id);
+  log(`⏳ Processing background task ${nextTask.id}...`);
+  try {
+    await processBackgroundTask(workspace, nextTask);
+  } catch (error) {
+    const errorMessage = extractErrorMessage(error);
+    const errorStack = error instanceof Error ? error.stack : undefined;
+    store.completeBackgroundTask(nextTask.id, errorMessage);
+    log(
+      `❌ Error processing background task ${nextTask.id}:\n${errorMessage}\n${errorStack}`
     );
+    return;
   }
+  store.completeBackgroundTask(nextTask.id);
+  log(`✅ Finished processing background task ${nextTask.id}`);
 }
 
-export function start({ store }: { store: Store }): void {
-  setTimeout(() => processBackgroundTasks(store));
+export function start({ workspace }: { workspace: Workspace }): void {
+  setTimeout(async () => {
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
```

Comment:
Avoids the "constant condition" lint:
```suggestion
    for (;;) {
```

Reply:
I'll stick with `while (true)` as per Jonah's line of reasoning